test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java

```diff
@@ -32,15 +38,33 @@
     private final Map<String, String> systemProps = new HashMap<>();
     private List<NativeImageStartedNotifier> startedNotifiers;
 
-    public NativeImageLauncher(Class<?> testClass) {
+    private NativeImageLauncher(Class<?> testClass, Config config) {
         this(testClass,
-                ConfigProvider.getConfig().getOptionalValue("quarkus.http.test-port", Integer.class).orElse(DEFAULT_PORT),
-                ConfigProvider.getConfig().getOptionalValue("quarkus.test.native-image-wait-time", Long.class)
-                        .orElse(DEFAULT_IMAGE_WAIT_TIME),
-                ConfigProvider.getConfig().getOptionalValue("quarkus.test.native-image-profile", String.class)
+                config.getValue("quarkus.http.test-port", OptionalInt.class).orElse(DEFAULT_PORT),
+                config.getValue("quarkus.test.native-image-wait-time", OptionalLong.class).orElse(DEFAULT_IMAGE_WAIT_TIME),
+                config.getOptionalValue("quarkus.test.native-image-profile", String.class)
                         .orElse(null));
     }
 
+    public NativeImageLauncher(Class<?> testClass) {
+        // todo: accessing run time config from here doesn't make sense
+        this(testClass, installAndGetSomeConfig());
+    }
+
+    private static Config installAndGetSomeConfig() {
+        final SmallRyeConfig config = ConfigUtils.configBuilder().build();
+        QuarkusConfigFactory.setConfig(config);
+        final ConfigProviderResolver cpr = ConfigProviderResolver.instance();
+        try {
+            final Config installed = cpr.getConfig();
+            if (installed != config) {
+                cpr.releaseConfig(installed);
```

Comment:
And you don't release the new config at the end?

Reply:
There didn't seem to be any need to do so.